nvtabular/io/dask.py

```diff
@@ -352,7 +353,10 @@ def _ddf_to_dataset(
             task_list.append(key)
         dsk[name] = (lambda x: x, task_list)
 
-    graph = HighLevelGraph.from_collections(name, dsk, dependencies=[ddf])
+    graph = _ensure_optimize_dataframe_graph(
+        dsk=HighLevelGraph.from_collections(name, dsk, dependencies=[ddf]),
```

Comment:
What's different about this call to `HighLevelGraph.from_collections()` from the one above in `_simple_shuffle()` that makes the optimization wrapper relevant here and not there? (Based on the name, I'd guess maybe `_simple_shuffle()` wouldn't benefit much from the optimizations?)

Reply:
Good question! 

In NVTabular, we only want to run an explicit optimization pass when we are about to `compute` something that is not a Dataframe collection.  For example, we want to make sure the graph is optimized before we compute statistics (which are derived from a collection, but not collections themselves).  We do not need to optimize when we merely construct a new collection, because (1) Dask will automatically run the optimizations if the collection itself is computed, and (2) the user may be performing further transforms on the collection anyway.

Reply:
Cool, thanks for the explanation!